app/src/main/java/com/cornellappdev/transit/ui/components/AddFavoritesSearchSheet.kt

```diff
@@ -0,0 +1,123 @@
+import com.cornellappdev.transit.ui.viewmodels.RouteViewModel
+
+/**
+ * Contents of AddFavorites BottomSheet
+ * @param routeViewModel The routeViewModel used in the app
+ * @param cancelOnClick The function to run when the cancel button is clicked
+ */
+@Composable
+fun AddFavoritesSearchSheet(
+    routeViewModel: RouteViewModel,
+    cancelOnClick: () -> Unit,
+) {
+
+
+    // Search bar flow
+    val searchBarValue = routeViewModel.searchQuery.collectAsState().value
+
+    Column(
+        modifier = Modifier
+            .fillMaxHeight(0.92f)
+            .fillMaxWidth()
+    ) {
+        Column(
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(color = Color.White),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(10.dp),
+                horizontalArrangement = Arrangement.SpaceBetween,
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Box(modifier = Modifier.weight(1f)) {
+                    Text(
+                        text = "Add Favorites",
+                        fontFamily = sfProDisplayFamily,
+                        fontStyle = FontStyle.Normal,
+                        fontWeight = FontWeight(600),
+                        fontSize = 20.sp,
+                        modifier = Modifier.align(Alignment.Center)
+                    )
+                    TextButton(
+                        onClick = cancelOnClick,
+                        content = {
+                            Text(
+                                text = "Cancel",
+                                fontFamily = sfProTextFamily,
+                                fontStyle = FontStyle.Normal,
+                                textAlign = TextAlign.Center,
+                                fontSize = 14.sp,
+                                fontWeight = FontWeight(500),
+                                color = TransitBlue,
+                            )
+                        },
+                        modifier = Modifier.align(Alignment.CenterEnd)
+                    )
+
+                }
+
+            }
+
+            SearchTextField(
+                value = searchBarValue,
+                setValue = { s -> routeViewModel.onQueryChange(s) },
+                placeholderText = "Search",
+                singleLine = true,
+                modifier = Modifier.padding(
+                    start = 16.dp,
+                    end = 16.dp,
+                    top = 7.dp,
+                    bottom = 11.dp
+                ),
+                height = 50.dp,
+                prefix = {
+                    Icon(
+                        Icons.Outlined.Search,
+                        "Search",
+                        modifier = Modifier.clickable { routeViewModel.onQueryChange("") })
```

Comment:
Is this the right logic? By what I can see, this will clear the search text if you click the search button? That sounds a bit strange.

Reply:
It's there rn as a placeholder since the actual search functionality isn't implemented yet. I'll add a comment!